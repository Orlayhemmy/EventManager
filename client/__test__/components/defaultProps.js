export const props = {
  userState: {
    user: {
      isAdmin: true
    },
    isAuth: true
  },
  location: {
    pathname: '/'
  },
  center: {
    status: '',
    message: ''
  }
};

export const center = {
  status: 201,
  message: ''
};

export const bookedEvent = [
  { id: 1, eventTitle: 'Pizza Party', bookedDate: [], isApproved: true },
  { id: 2, eventTitle: 'Wedding Ceremony', bookedDate: [], isApproved: true  },
  { id: 3, eventTitle: 'Seminar', bookedDate: [], isApproved: false  }
];

export const logout = jest.fn();
export const onChange = jest.fn();
