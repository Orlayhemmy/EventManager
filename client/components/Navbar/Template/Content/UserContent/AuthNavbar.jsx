import React from 'react';
import { Link } from 'react-router-dom';

export default class AuthNavbar extends React.Component {
  /**
   * @memberof NavBar
   * @method logout
   * @description it calls a logout action
   * @param {object} event
   * @returns {void}
   */
  logout(e) {
    this.props.navbarProps.logout();
  }
  /**
   * @memberof AuthNavbar
   * @method render
   * @description it renders the content of authNavbar
   * @returns the HTML of NavBar
   */
  render() {
    const navbarSecondary = (
      <ul className="navbar-nav ml-auto">
        <li className="nav-item">
          <Link to="/dashboard" className="nav-link">
            <span className="nav-link-text">
              <i className="fa fa-dashboard" /> Dashboard
            </span>
          </Link>
        </li>
        <li className="nav-item">
          <Link to="/admin-centers" className="nav-link">
            <span className="nav-link-text">
              <i className="fa fa-home" /> centers
            </span>
          </Link>
        </li>
        <li className="nav-item">
          <Link to="/add-center" className="nav-link">
            <span className="nav-link-text">
              <i className="fa fa-plus-square" /> add center
            </span>
          </Link>
        </li>
        <li className="nav-item">
          <Link to="/add-event" className="nav-link">
            <span className="nav-link-text">
              <i className="fa fa-plus-square" /> add event
            </span>
          </Link>
        </li>
        <li className="nav-item">
          <Link to="/profile" className="nav-link">
            <span className="nav-link-text">
              <i className="fa fa-user" /> profile
            </span>
          </Link>
        </li>
        <li className="nav-item">
          <Link to="/" onClick={this.logout.bind(this)} className="nav-link">
            <span className="nav-link-text">
              <i className="fa fa-power-off" /> logout
            </span>
          </Link>
        </li>
      </ul>
    );

    const navbarMain = (
      <ul className="navbar-nav ml-auto">
        <li className="nav-item">
          <Link to="/" className="nav-link">
            <span className="nav-link-text">
              <i className="fa fa-home" /> Home
            </span>
          </Link>
        </li>
        <li className="nav-item">
          <Link to="/about" className="nav-link">
            <span className="nav-link-text">
              <i className="fa fa-dashboard" /> about us
            </span>
          </Link>
        </li>
        <li className="nav-item">
          <Link to="/dashboard" className="nav-link">
            <span className="nav-link-text">
              <i className="fa fa-dashboard" /> Dashboard
            </span>
          </Link>
        </li>
        <li className="nav-item">
          <Link
            to="/"
            onClick={this.logout.bind(this)}
            className="nav-link"
          >
            <span className="nav-link-text">
              <i className="fa fa-power-off" /> logout{' '}
            </span>
          </Link>
        </li>
      </ul>
    );
    return (
      <div>
        {this.props.path === '/' || this.props.path === '/about' ?
          navbarMain : navbarSecondary }
      </div>
    );
  }
}
