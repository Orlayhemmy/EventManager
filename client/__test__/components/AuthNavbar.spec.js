import React from 'react';
import AuthNavbar from '../../components/Navbar/Template/Content/UserContent/AuthNavbar';
import { navbarProps, logout } from './defaultProps';

describe('Auth navbar', () => {
  const wrapper = shallow(<AuthNavbar navbarProps={navbarProps} logout={logout}/>);
  it('should render the component without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
