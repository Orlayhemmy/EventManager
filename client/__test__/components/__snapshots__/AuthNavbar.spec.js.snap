// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth navbar should render the component without errors 1`] = `
<ul
  className="navbar-nav ml-auto"
>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link active"
      replace={false}
      to="/"
    >
      <span
        className="nav-link-text"
      >
        Home
      </span>
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      replace={false}
      to="/about"
    >
      <span
        className="nav-link-text"
      >
        about us
      </span>
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      replace={false}
      to="/dashboard"
    >
      <span
        className="nav-link-text"
      >
        Dashboard
      </span>
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      onClick={[Function]}
      replace={false}
      to="/"
    >
      <span
        className="nav-link-text"
      >
        logout
      </span>
    </Link>
  </li>
</ul>
`;
